ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/Divulgence.scala

```diff
@@ -0,0 +1,133 @@
+          )
+
+          assert(
+            createDivulgence2.getCreated.contractId == divulgence2.contractId,
+            s"The event where Divulgence2 is created should have the same contract identifier as the created contract"
+          )
+
+          val exerciseOnDivulgence2Transaction = bobTrees(1)
+          assert(
+            exerciseOnDivulgence2Transaction.rootEventIds.size == 1,
+            s"The transaction where a choice is exercised on Divulgence2 should contain exactly one root event contains ${exerciseOnDivulgence2Transaction.rootEventIds.size} instead"
+          )
+
+          val exerciseOnDivulgence2 = exerciseOnDivulgence2Transaction.eventsById(
+            exerciseOnDivulgence2Transaction.rootEventIds(0))
+          assert(
+            exerciseOnDivulgence2.kind.isExercised,
+            s"Expected event to be an exercise"
+          )
+
+          assert(exerciseOnDivulgence2.getExercised.contractId == divulgence2.contractId)
+
+          assert(exerciseOnDivulgence2.getExercised.childEventIds.size == 1)
+
+          val exerciseOnDivulgence1 =
+            exerciseOnDivulgence2Transaction.eventsById(
+              exerciseOnDivulgence2.getExercised.childEventIds(0))
+
+          assert(exerciseOnDivulgence1.kind.isExercised)
+
+          assert(exerciseOnDivulgence1.getExercised.contractId == divulgence1.contractId)
+
+          assert(exerciseOnDivulgence1.getExercised.childEventIds.isEmpty)
+
+          assert(
+            aliceTransactions.size == 3,
+            s"Alice should see three transactions but sees ${aliceTransactions.size} instead")
+
+          assert(aliceTransactions.head.events.size == 1)
+
+          assert(aliceTransactions.head.events.head.event.isCreated)
+
+          assert(
+            aliceTransactions.head.events.head.event.created.get.contractId == divulgence1.contractId)
+
+          assert(aliceTransactions.head.events.head.event.created.get.witnessParties == Seq(alice))
+        }
+    }
+
+  private val activeContractServiceDivulgence = {
+    LedgerTest("Divulged contracts should not be exposed by the active contract service") {
+      implicit context =>
+        for {
+          Vector(alice, bob) <- allocateParties(2)
+          divulgence1 <- Divulgence1(alice, alice)
+          divulgence2 <- Divulgence2(bob, bob, alice)
+          _ <- divulgence2.fetch(controller = alice, divulgence1)
+          activeForBob <- activeContracts(bob)
+          activeForAlice <- activeContracts(alice)
+        } yield {
+          assert(activeForBob.size == 1)
+          assert(activeForBob.head.contractId == divulgence2.contractId)
+          assert(activeForBob.head.witnessParties == Seq(bob))
+          assert(activeForAlice.size == 2)
+          assert(
+            activeForAlice.map(_.contractId) == Seq(divulgence1.contractId, divulgence2.contractId))
+          assert(activeForAlice(0).witnessParties == Seq(alice))
+          assert(activeForAlice(1).witnessParties == Seq(alice))
```

Comment:
Same comment as above. Let's preserve the behavior of the test with respect to both bob and alice requesting transactions.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/2282/commits/c7e6538d4efaacb3ed283c726acc0a12dfaaa80f